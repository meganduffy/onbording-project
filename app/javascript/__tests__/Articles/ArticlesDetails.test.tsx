import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import ArticleDetails from '../../components/Articles/ArticlesDetails';

let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', ()=>{
  const tree = renderer.create(<ArticleDetails 
    title='New Post' 
    content='Ipsum Lorem'
    firstName='Megan' 
    link='google.com' 
    linkText='Click Here' 
    created_at='27 March 2020' 
    editLink='/articles/3/edit'/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('displays an edit option when logged in', ()=>{
  const tree = renderer.create(<ArticleDetails 
    title='New Post' 
    content='Ipsum Lorem'
    firstName='Megan' 
    link='google.com' 
    linkText='Click Here' 
    created_at='27 March 2020' 
    editLink='/articles/3/edit'
    currentUser={2}/>).toJSON();
    expect(tree).toMatch(/Edit Article/)
})
