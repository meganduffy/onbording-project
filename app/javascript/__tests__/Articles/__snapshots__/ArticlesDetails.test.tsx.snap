// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetails Snapshots should render the Article Details when logged in 1`] = `
<div>
  <div
    class="outer-article-container"
    data-testid="article-details"
  >
    <div
      class="articles-details"
    >
      <h2
        class="articles-title"
      >
        mechanism momentum museum minimum mushroom
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <div
        class="articles-details-info"
      >
        <p>
          Megan
           
          Duffy
        </p>
        <p>
          27 March 2020
        </p>
      </div>
    </div>
    <div
      class="articles-aside"
    >
      <a
        href="/articles"
      >
        View All Articles
      </a>
      <a
        href="/articles/3/edit"
      >
        Edit Article
      </a>
      <a>
        Delete Article
      </a>
    </div>
  </div>
</div>
`;

exports[`ArticleDetails Snapshots should render the Article Details when logged out 1`] = `
<div>
  <div
    class="outer-article-container"
    data-testid="article-details"
  >
    <div
      class="articles-details"
    >
      <h2
        class="articles-title"
      >
        mechanism momentum museum minimum mushroom
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <div
        class="articles-details-info"
      >
        <p>
          Megan
           
          Duffy
        </p>
        <p>
          27 March 2020
        </p>
      </div>
    </div>
    <div
      class="articles-aside"
    >
      <a
        href="/articles"
      >
        View All Articles
      </a>
    </div>
  </div>
</div>
`;
